addons/html_editor/static/src/main/link/link_popover.js

```diff
@@ -347,6 +357,31 @@ export class LinkPopover extends Component {
             return deduceURLfromText(text, this.props.linkElement) || "";
         }
     }
+    getButtonShape() {
+        const shapeToRegex = (shape) => {
+            const parts = shape.trim().split(/\s+/);
+            const regexParts = parts.map((cls) => {
+                if (["outline", "fill"].includes(cls)) {
+                    cls = `btn-${cls}`;
+                }
+                return `(?=.*\\b${cls}\\b)`;
+            });
+            return new RegExp(regexParts.join(""));
+        };
+        // If multiple shapes match, prefer the one with more specificity.
+        let shapeMatched = "";
+        let matchScore = 0;
+        for (const { shape } of this.buttonShapeData) {
+            const regex = shapeToRegex(shape);
+            if (regex.test(this.props.linkElement.className)) {
+                if (matchScore < shape.length) {
+                    matchScore = shape.length;
+                    shapeMatched = shape;
+                }
```

Comment:
What I meant by the number of shapes is the number of parts. 
For example, ‘fill out’ = 2, 
“fill” = 1 
‘fill out plop’ = 3